lib/clone/index.js

```diff
@@ -1,45 +1,65 @@
 import __ from '../__'
-import findIndex from '../findIndex/findIndex'
+import identity from '../identity'
+import indexOf from '../indexOf/indexOf'
 
-function baseClone(src, circulars, clones) {
+function cloneRegExp(pattern) {
+  var flags = ''
+  if (pattern.global) { flags += 'g' }
+  if (pattern.ignoreCase) { flags += 'i' }
+  if (pattern.multiline) { flags += 'm' }
+  if (pattern.sticky) { flags += 'y' }
+  if (pattern.unicode) { flags += 'u' }
+  return new RegExp(pattern.source, flags)
+}
+
+function baseClone(src, circulars, clones, deep) {
   if (src === __) return __
 
   if (!src || typeof src !== 'object' || typeof src === 'function') {
     return src
   }
 
   if (src.nodeType && 'cloneNode' in src) {
-    return src.cloneNode(true)
+    return src.cloneNode(deep)
   }
 
   if (src instanceof Date) {
     return new Date(src.getTime())
   }
 
   if (src instanceof RegExp) {
-    return new RegExp(src)
+    return cloneRegExp(src)
   }
 
   if (Array.isArray(src)) {
-    return src.map(clone)
+    return src.map(deep ? function(value) {
+      return clone(value, true)
+    } : identity)
```

Comment:
Could reduce size for non-deep cloning a little when removing `indentity`, because of less dependencies.

```js
return src.map(function(value) {
  return deep ? clone(value, true) : value
})
```

Also, is this variant has acceptable behavior, or you need explicit 1st level shallow cloning for non-deep? :

```js
return src.map(function(value) {
  return clone(value, deep)
})
```

Reply:
Remove `identity` is a good point here, agreed 👍 -12B

Second suggestion gives incorrect behaviour and -4B 📛 
Incorrect is because of `clone` deep option, anyway it clones its values which is not expected